File: backend/src/main/scala/shared/objects/response/ErrorResponse.scala
package shared.objects.response

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
import shared.objects.ApiMessageParams

final case class ErrorResponse(
  code: Option[String],
  message: Option[String],
  params: ApiMessageParams
)

object ErrorResponse:
  given Encoder[ErrorResponse] = deriveEncoder[ErrorResponse]
  given Decoder[ErrorResponse] = deriveDecoder[ErrorResponse]

File: frontend/src/objects/shared/transport/ErrorResponse.ts
import type { ApiMessageParams } from '@/objects/shared/ApiMessageParams'

/** API 错误响应；code/message 可为空，params 用于国际化或模板化错误文案。 */
export type ErrorResponse = {
  code: string | null
  message: string | null
  params: ApiMessageParams
}
